www/guides/react.md

```diff
@@ -0,0 +1,373 @@
+
+And it also tells it to put the final output files in a subdirectory called `_dist_` so change that path in `index.html` from `/index.js` to `/_dist_/index.js`.
+
+```diff
+    <h1>Welcome to Snowpack!</h1>
+    <div id="root"></div>
+-    <script type="module" src="/index.js"></>
++    <script type="module" src="/_dist_/index.js"></script>
+  </body>
+```
+
+If Snowpack is still running, exit the process, you need to restart Snowpack for configuration file changes. When you start up again, if it worked, it should look the same.
+
+Create a new file at `src/App.jsx` and paste the following code into this new file to create an `App` component:
+
+```jsx
+import React, {useState, useEffect} from 'react';
+
+function App() {
+  // Create the count state.
+  const [count, setCount] = useState(0);
+  // Create the counter (+1 every second).
+  useEffect(() => {
+    const timer = setTimeout(() => setCount(count + 1), 1000);
+    return () => clearTimeout(timer);
+  }, [count, setCount]);
+  // Return the App component.
+  return (
+    <div className="App">
+      <header className="App-header">
+        <p>
+          Edit <code>src/App.jsx</code> and save to reload.
+        </p>
+        <p>
+          Page has been open for <code>{count}</code> seconds.
+        </p>
+        <p>
+          <a
+            className="App-link"
+            href="https://reactjs.org"
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            Learn React
+          </a>
+        </p>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+Now include it in `index.js`
+
+```diff
+import React from 'react';
+import ReactDOM from 'react-dom';
++ import App from './App';
+
+- ReactDOM.render(<div>"HELLO WORLD"</div>, document.getElementById('root'));
++ ReactDOM.render(
++    <React.StrictMode>
++      <App />
++    </React.StrictMode>,
++    document.getElementById('root'),
++  );
+```
+
+> [Strict Mode](https://reactjs.org/docs/strict-mode.html) is a tool for highlighting potential problems in React code.
+
+You shouldn't need to restart Snowpack to see this, it should look like this:
+
+IMAGE: screenshot of the project, showing text saying "edit `src/App.js` and save to reload," text that says "Page has been open for" and the number of seconds then "seconds". A link with the text Learn React that links to the React website.
+
+## Styling your Snowpack/React project
+
+When you add assets like images or CSS, Snowpack includes them in your final build. If you already know React, this process should look pretty familiar.
+
+> As you're doing this, you should not need to reload the page or restart Snowpack. Snowpack updates the project in the browser as you edit code, which is pretty nifty.
+
+Add this file `logo.svg` (link to file) to `src`. Now you can import it into your `App.jsx` and use it in an `img` tag to display it.
```

Comment:
shouldn't assets like `logo.svg` be in `public`? earlier in the guide it said

> Then a public directory for static assets like images, fonts, icons, and sometimes CSS.

Reply:
Good question, I'll leave this one open. I think sometimes if assets are component specific people will add them in the component directory.

Reply:
+1 for @melissamcewen's answer here, although that's a good point of possible confusion. Maybe worth clarifying with an added line to this section like "for some static assets, you may want to locate them near the React components that use them..."

and clarifying the public line with something like "a public directory for **site** assets like images, fonts, icons, and sometimes CSS."